daemon/logger/jsonfilelog/jsonfilelog.go

```diff
@@ -124,72 +124,85 @@ func (l *JSONFileLogger) Log(msg *logger.Message) error {
 }
 
 func writeLog(l *JSONFileLogger) (int64, error) {
+
+	if err := checkCapacityAndRotate(l); err != nil {
+		return -1, err
+	}
+
+	i, err := l.buf.WriteTo(l.f)
+	if err != nil {
+		l.buf.Reset()
```

Comment:
If During process WriteTo error occur, WriteTo still clean the buffer?(for example if writeTo return at the line 213 or 218.. in your link)